homeassistant/components/homekit/config_flow.py

```diff
@@ -0,0 +1,288 @@
+from homeassistant.helpers.entityfilter import (
+    CONF_EXCLUDE_DOMAINS,
+    CONF_EXCLUDE_ENTITIES,
+    CONF_INCLUDE_DOMAINS,
+    CONF_INCLUDE_ENTITIES,
+)
+
+from .const import (
+    CONF_AUTO_START,
+    CONF_FILTER,
+    CONF_SAFE_MODE,
+    CONF_ZEROCONF_DEFAULT_INTERFACE,
+    DEFAULT_AUTO_START,
+    DEFAULT_CONFIG_FLOW_PORT,
+    DEFAULT_SAFE_MODE,
+    DEFAULT_ZEROCONF_DEFAULT_INTERFACE,
+    SHORT_BRIDGE_NAME,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+from .util import find_next_available_port
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DOMAINS = "domains"
+SUPPORTED_DOMAINS = [
+    "alarm_control_panel",
+    "automation",
+    "binary_sensor",
+    "climate",
+    "cover",
+    "demo",
+    "device_tracker",
+    "fan",
+    "input_boolean",
+    "light",
+    "lock",
+    "media_player",
+    "person",
+    "remote",
+    "scene",
+    "script",
+    "sensor",
+    "switch",
+    "vacuum",
+    "water_heater",
+]
+
+DEFAULT_DOMAINS = [
+    "alarm_control_panel",
+    "climate",
+    "cover",
+    "light",
+    "lock",
+    "media_player",
+    "switch",
+    "vacuum",
+    "water_heater",
+]
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for HomeKit."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize config flow."""
+        self.homekit_data = {}
+        self.entry_title = None
+
+    async def async_step_pairing(self, user_input=None):
+        """Pairing instructions."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.entry_title, data=self.homekit_data
+            )
+        return self.async_show_form(
+            step_id="pairing",
+            description_placeholders={CONF_NAME: self.homekit_data[CONF_NAME]},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            port = await self._async_available_port()
+            name = self._async_available_name()
+            title = f"{name}:{port}"
+            self.homekit_data = user_input.copy()
+            self.homekit_data[CONF_NAME] = name
+            self.homekit_data[CONF_PORT] = port
+            self.homekit_data[CONF_FILTER] = {
+                CONF_INCLUDE_DOMAINS: user_input[CONF_INCLUDE_DOMAINS],
+                CONF_INCLUDE_ENTITIES: [],
+                CONF_EXCLUDE_DOMAINS: [],
+                CONF_EXCLUDE_ENTITIES: [],
+            }
+            del self.homekit_data[CONF_INCLUDE_DOMAINS]
+            self.entry_title = title
+            return await self.async_step_pairing()
+
+        default_domains = [] if self._async_current_entries() else DEFAULT_DOMAINS
+        setup_schema = vol.Schema(
+            {
+                vol.Optional(CONF_AUTO_START, default=DEFAULT_AUTO_START): bool,
```

Comment:
Can we only add this option if `self.show_advanced_options` ? With the `started` event combined with entity registry restoring unavailable entities, it's not necessary anymore

Reply:
Done, looks the the frontend (https://github.com/home-assistant/frontend/pull/5612) isn't merged yet so I wrote coverage for the basic path

Reply:
Yeah Bram was waiting until we didn't need frontend releases done for 109 anymore.